After I figure out how to access proper Jenkins build page I see failed test, but I have no idea how to find those which are new. Can it be done or should developer come at every single one to scan for errors? The page reports (74 failures / +3) which I read as 3 new failed tests, but I have no idea which ones.

For example, if I click on first one (WMCore_t.WMBS_t.Job_t.JobTest.testJobCacheDir) I see this error:
```
root: DEBUG: Using SQLAlchemy v.1.3.3
root: INFO: Instantiating base WM DBInterface
root: ERROR: Critical error while attempting to delete entire DB!
(exceptions.LookupError) unknown encoding: utf8mb4
[SQL: SELECT CAST('test plain returns' AS CHAR(60)) AS anon_1]Traceback (most recent call last):
```
but from test details I see that it is python 2.7 build, should we care? The tests itself is not related `Common.py` module I'm changing. Therefore I can conclude that this failure has nothing to do with changes I provide.

The only useful insight comes from [Py3 Unit tests](https://cmssdt.cern.ch/dmwm-jenkins/view/All/job/DMWM-WMCore-PR-test/12734/artifact/artifacts/PullRequestReport.html#unittestspy3) page where in Python2 pylint summary it claims:
```
E0611, line 29: No name 'aggRuns' in module 'dbs.apis.dbsClient'
E0611, line 29: No name 'aggFileLumis' in module 'dbs.apis.dbsClient'
```
which tells me that DBSClient version is wrong since it does not provide this functions. Which brings me to the question which env is used in Unit tests and how I can verify that DBSClient there is up to date?

--- a/src/python/WMCore/MicroService/Tools/Common.py
+++ b/src/python/WMCore/MicroService/Tools/Common.py
@@ -25,11 +25,30 @@ from Utils.CertTools import getKeyCertFromEnv
 from WMCore.Services.pycurl_manager import RequestHandler
 from WMCore.Services.pycurl_manager import getdata as multi_getdata
 
+# DBS agregators
+from dbs.apis.dbsClient import aggRuns, aggFileLumis
+
 
 # static variables
 STEP_PAT = re.compile(r'Step[0-9]')
 TASK_PAT = re.compile(r'Task[0-9]')
 
+
+def isEmptyResults(row):
+    """
+    _isEmptyResults_
+
+    Evaluates whether row data contains empty result set
+    :return: bool
+    """
+    if 'data' not in row:
+        raise Exception("provided result dict does not contain 'data' key")
+    # if code is not present in row it means it was success (HTTP status code 200)
+    code = int(row.get('code', 200))
+    data = row['data']
+    if (code >= 200 and code < 400) and data in (None, []):
+        return True
+    return False
 
 def getMSLogger(verbose, logger=None):
     """
@@ -63,7 +82,7 @@ def dbsInfo(datasets, dbsUrl):
 
     for row in data:
         dataset = row['url'].split('=')[-1]
-        if row['data'] is None:
+        if isEmptyResults(row):
             print("FAILURE: dbsInfo for %s. Error: %s %s" % (dataset, row.get('code'), row.get('error')))
             continue
         rows = json.loads(row['data'])
@@ -232,7 +251,7 @@ def getBlocksByDsetAndRun(datasetName, runList, dbsUrl):
 
     for row in data:
         dataset = row['url'].rsplit('=')[-1]
-        if row['data'] is None:
+        if isEmptyResults(row):
             msg = "Failure in getBlocksByDsetAndRun for %s. Error: %s %s" % (dataset,
                                                                              row.get('code'),
                                                                              row.get('error'))
@@ -261,12 +280,13 @@ def getFileLumisInBlock(blocks, dbsUrl, validFileOnly=1):
 
     for row in data:
         blockName = unquote(row['url'].rsplit('=')[-1])
-        if row['data'] is None:
+        if isEmptyResults(row):
             msg = "Failure in getFileLumisInBlock for block %s. Error: %s %s" % (blockName,
                                                                                  row.get('code'),
                                                                                  row.get('error'))
             raise RuntimeError(msg)
         rows = json.loads(row['data'])
+        rows = aggFileLumis(rows) # adjust to DBS Go server output
         runLumisByBlock.setdefault(blockName, [])
         for item in rows:
             runLumisByBlock[blockName].append(item)
@@ -288,7 +308,7 @@ def findBlockParents(blocks, dbsUrl):
     for row in data:
         blockName = unquote(row['url'].rsplit('=')[-1])
         dataset = blockName.split("#")[0]
-        if row['data'] is None:
+        if isEmptyResults(row):
             print("Failure in findBlockParents for block %s. Error: %s %s" % (blockName,
                                                                               row.get('code'),
                                                                               row.get('error')))
@@ -322,12 +342,13 @@ def getRunsInBlock(blocks, dbsUrl):
     data = multi_getdata(urls, ckey(), cert())
     for row in data:
         blockName = unquote(row['url'].rsplit('=')[-1])
-        if row['data'] is None:
+        if isEmptyResults(row):
             msg = "Failure in getRunsInBlock for block %s. Error: %s %s" % (blockName,
                                                                             row.get('code'),
                                                                             row.get('error'))
             raise RuntimeError(msg)
         rows = json.loads(row['data'])
+        rows = aggRuns(rows) # adjust to DBS Go server output
         runsByBlock[blockName] = rows[0]['run_num']
     return runsByBlock
 
@@ -366,7 +387,7 @@ def eventsLumisInfo(inputs, dbsUrl, validFileOnly=0, sumOverLumi=0):
     data = multi_getdata(urls, ckey(), cert())
     for row in data:
         data = unquote(row['url'].split('=')[-1])
-        if row['data'] is None:
+        if isEmptyResults(row):
             print("FAILURE: eventsLumisInfo for %s. Error: %s %s" % (data,
                                                                     row.get('code'),
                                                                     row.get('error')))
@@ -585,7 +606,7 @@ def findParent(datasets, dbsUrl):
 
     for row in data:
         dataset = row['url'].split('=')[-1]
-        if row['data'] is None:
+        if isEmptyResults(row):
             print("Failure in findParent for dataset %s. Error: %s %s" % (dataset,
                                                                           row.get('code'),
                                                                           row.get('error')))

--- a/test/python/WMCore_t/MicroService_t/Tools_t/Common_t.py
+++ b/test/python/WMCore_t/MicroService_t/Tools_t/Common_t.py
@@ -7,7 +7,8 @@ from __future__ import division, print_function
 
 import unittest
 
-from WMCore.MicroService.Tools.Common import dbsInfo, getEventsLumis, findParent
+from WMCore.MicroService.Tools.Common import \
+        dbsInfo, getEventsLumis, findParent, isEmptyResults
 
 
 class CommonTest(unittest.TestCase):
@@ -47,6 +48,19 @@ class CommonTest(unittest.TestCase):
         parents = findParent(self.child, self.dbsUrl)
         self.assertEqual(parents[self.child[0]], '/SingleElectron/Run2016B-v2/RAW')
 
+    def test_isEmptyResults(self):
+        row = {'code': 200, 'data': None}
+        result = isEmptyResults(row)
+        self.assertEqual(result, True)
+        row = {'code': 200, 'data': []}
+        result = isEmptyResults(row)
+        self.assertEqual(result, True)
+        row = {'code': 400, 'data': None}
+        result = isEmptyResults(row)
+        self.assertEqual(result, False)
+        row = {'code': 400, 'data': []}
+        result = isEmptyResults(row)
+        self.assertEqual(result, False)
 
 if __name__ == '__main__':
     unittest.main()

